Interfaces/API/Dirac.py

```diff
@@ -2600,7 +2600,7 @@ def __getJDLParameters( self, jdl ):
     :param jdl: a JDL
     :type jdl: ~DIRAC.Interfaces.API.Job.Job or str or file
     """
-    if isinstance(jdl, DIRAC.Interfaces.API.Job.Job):
+    if hasattr(jdl, '_toJDL'):
```

Comment:
What's `jdl` in your case? Also, maybe then you need to change the docstring of this method.

Reply:
just using more pythonic duck typing to figure out which path to follow. Still only Dirac Job and sub classes should have this function, there could be issues when the import for Job comes from some other way.